### PR TITLE
Rtk query refactoring

### PR DESCRIPTION
**Pull request by hys25**

### CODE REVIEW

File: src/pages/project/ProjectView.js [hunk 1]
```diff
@@ -36,11 +29,13 @@ function ProjectView() {
   )
   const handleRemoveProject = useCallback(async () => {
     setOpenModal(false)
-    const result = await dispatch(deleteProject(currentProjectId))
-    if (result.payload.status === 200) {
+    const result = await deleteProject(currentProjectId)
+    if (result.data.id === currentProjectId) {
       navigate("/")
+    } else {
+      toast.error("Project was deleted")
```
Comment (ArturHorbenko) [inline src/pages/project/ProjectView.js]: I think in this case project was **not** deleted

File: src/config/errorHandlerMiddeware.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { isRejectedWithValue } from "@reduxjs/toolkit"
+import { toast } from "react-toastify"
+
+export const rtkQueryErrorLogger = () => (next) => (action) => {
+  if (isRejectedWithValue(action)) {
```
Comment (ArturHorbenko) [inline src/config/errorHandlerMiddeware.js]: We can extend a little our error handling here. 
There is also `isRejected` ( https://redux-toolkit.js.org/api/matching-utilities#isrejected ) in rtk-query, which indicates if promise was rejected. The one we use is only when promise is rejected with value. 

I propose next, we leave `isRejectedWithValue` as it is and we add additional check with `isRejected` down below and we show some useful info like what url, what status code and what error are in these case ( if there are any )

So if we for example will get rejected promise with `data.message` : "No Id found", we will display this. 
If we just get rejected promise without any value, we will anyway show some general error, request url and status code in this case.

To test it, we can try to modify some request on backend to reject without any body.

File: src/features/auth/authService.js
Comment (ArturHorbenko) [inline src/features/auth/authService.js:1]: I think we still should call this file authApi for consistency across the project.

**Review by ArturHorbenko — COMMENTED** [review]
Hey, you did a great job on reducing boilerplate and modernizing admin tool !
I added couple minor comments to PR, once it is resolved we can look into deployment.

**Comment by hys25** [comment]
> Hey, you did a great job on reducing boilerplate and modernizing admin tool ! I added couple minor comments to PR, once it is resolved we can look into deployment.

thanks, pushed changes

**Pull request merged**